### PR TITLE
vits-bigvgan

### PR DESCRIPTION
**Pull request by manmay-nakhashi**
added option of using bigvgan as a waveform decoder in vits 
added seperate bigvgan_generator and discriminator if anyone wants to train it seperately 
added vits_bigvgan recipe 
added test cases
code is implemented based on => https://github.com/NVIDIA/BigVGAN

### CODE REVIEW

**Comment by manmay-nakhashi** [comment]
bigvgan-lj-1
bigvgan-lj-2

**Comment by manmay-nakhashi** [comment]
https://soundcloud.com/manmay-nakhashi/vits-bigvgan-1?si=9c440e824fb34e9689e73f2525d24f1f&utm_source=clipboard&utm_medium=text&utm_campaign=social_sharing

**Comment by erogol** [comment]
Thanks for the PR. I have some questions.

How does it compare to the original VITS? Is there any clear improvement? 

Did you find a workaround to the loss jump issue you were mentioning on Discord?

It'd be nice to have a vocoder-only recipe too, to support the other TTS models.

File: TTS/tts/models/vits.py [hunk 1]
```diff
@@ -596,6 +597,19 @@ class VitsArgs(Coqpit):
     reinit_DP: bool = False
     reinit_text_encoder: bool = False
 
+    # bigvgan params
```
Comment (erogol) [inline TTS/tts/models/vits.py:600]: Can you add a reference if it corresponds to a model mentioned in the paper?

**Comment by manmay-nakhashi** [comment]
How does it compare to the original VITS? Is there any clear improvement?
- i have only trained right now on ljspeech due to gpu unavailabilty , haven't trained on multispeaker model to check the zero-shot output which they have mentioned in the paper.(ljspeech training in progress trained 100k steps i feel treble has been improved from original vits ) - i'll post the samples after 200k steps right now 100k steps
- we need to test this on voice conversion and zero-shot tts because in peper they claim better results for unseen wavs. 

Did you find a workaround to the loss jump issue you were mentioning on Discord?
- We found that the default learning rate of 2×10−4 used in HiFi-GAN causes an early training collapse for BigVGAN training, where the losses from the discriminator submodules immediately converge to zero after several thousands of iterations. Halving the learning rate to 1 × 10−4 was able to reduce such failures. We also found that large batch size is helpful to reduce mode collapse at training (Brock et al., 2019). We only double the batch size from the usual 16 to 32 for a good trade-off between training efficiency and stability, as neural vocoders can require millions of steps to converge. Note that this recommended batch size is still much smaller than the one for image synthesis (e.g., 2048) (Brock et al., 2019), because neural vocoding has strong conditional information. (loss jump was happening becasue of the same reason )
It'd be nice to have a vocoder-only recipe too, to support the other TTS models.
-i'll add the recipies and it's test cases in the same pr.

**Comment by manmay-nakhashi** [comment]
loss comparision by learning rate
[vits-bigvgan_lr_1e-4.docx](https://github.com/coqui-ai/TTS/files/10899043/vits-bigvgan_lr_1e-4.docx)
[vits-bigvgan_lr_default.odt](https://github.com/coqui-ai/TTS/files/10899044/vits-bigvgan_lr_default.odt)

**Comment by manmay-nakhashi** [comment]
added independent scripts for bigvgan vocoder training.
added vctk scripts for vits-bigvgan.

**Comment by manmay-nakhashi** [comment]
@erogol i haven't changed the base_tacotron.py but somehow style check is failing on that.
